core/FrontController.php

```diff
@@ -402,7 +402,14 @@ public function init()
         // ... if session auth fails try normal auth (which will login the anonymous user)
         if (!$loggedIn) {
             $authAdapter = $this->makeAuthenticator();
-            Access::getInstance()->reloadAccess($authAdapter);
+            try {
+                Access::getInstance()->reloadAccess($authAdapter);
+            } catch (\Zend_Db_Statement_Exception $e) {
+                // table `user_token_auth` might not yet exist when updating to Matomo 4
+                if (!strpos($e->getMessage(), 'user_token_auth')) {
```

Comment:
Would maybe need to additionally check whether the user_token_auth table exists or not for additional security. Ideally maybe the query in the user model (not sure which model is used there) would catch this maybe and on exception check for the text and whether table exists?  Technically could additionally even check the mysql error code maybe? (although that should not be too important)

Reply:
@tsteur Haven't looked at the code in detail yet. But does the anonymous user need to have a token_auth in the table at all? Wondering if we couldn't simply return `anonymous` as token_auth for the `anonymous` user always 🤔

Reply:
Tested locally and noticed w/ this code `Access::getInstance()->getSitesIdWithViewAccess()` always returns an empty array. In this case, we could also add a check that the sites w/ access is empty, that might be secure enough? Since we make sure the logged in user (which is no one, not even anonymous w/ this code) has basically no access.

Reply:
> Wondering if we couldn't simply return anonymous as token_auth for the anonymous user always

That should work too.